### ISSUE TITLE
Firestore.instance.collection(path).stream stop working after changing WiFi network  

### ISSUE THREAD

**Issue by guyluz11** [cachapa/firedart#64]
I am initializing Firestore and calling `Firestore.instance.collection(path).stream` and it is working.

If I change network while the app is running and calling `Firestore.instance.collection(path).stream` again new collection values are not arriving when being changed.

The last stream was canceled before the new one was being called.

Is there part of the library that fixed on the network that he got when it was initialized like the httpClient?.

**Comment by kekko7072** [comment]
I have the same issue running the app on a RPi with flutter-pi, but I have this problem without changing the wifi. My code is 
```
StreamBuilder<....>(
        stream: firebaseCollection.document(docName!).stream,
        builder: (context, snapshot) {
          if (snapshot.hasData) {
      ////
}
else {
            return Center(child: CircularProgressIndicator());
          }
        },
      ),
```
Any suggestion on the reason why?

**Comment by guyluz11** [comment]
If I understand you correctly the stream stop working for you after number of hours without being closed and without showing any error?.

If so than I think that this is **different issue** than this one and you should open new issue about it.

**Comment by pritam-explorex** [comment]
Guys any update on this ?

**Comment by guyluz11** [comment]
> Guys any update on this ?

I have an [open a pr](https://github.com/cachapa/firedart/pull/67) that **might** solve this, can you test if you can reproduce this in my fork as well?
 
https://github.com/guyluz11/firedart

**Comment by stabjul** [comment]
It's still happening, streams close without any msg/exception/error. I cloned your code into my project and ran it (cross-checked if your changes were present - from the PR). 

Is there a workaround for this issue?

**Comment by cachapa** [comment]
Closed by #130
